cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs

```diff
@@ -3181,9 +3183,24 @@ pMaxTransactionSize =
       , Opt.help "Maximum transaction size."
       ]
 
+wordReader :: (Integral a, Bits a) => String -> ReadM a
+wordReader typeString =
+  Opt.maybeReader parser
+ where
+  -- We use 'readMaybe' instead of 'read', because the latter wouldn't output a nice error message
+  parser s =
+    case readMaybe s >>= naturalToWordMaybe >>= toIntegralSized of
+      Nothing ->
+        error $ "Cannot parse " <> s <> " as a " <> typeString
+      Just a ->
+        Just a
+
+word16Reader :: ReadM Word16
+word16Reader = wordReader @Word16 "Word16"
```

Comment:
We could use fancy type-level stuff to avoid this `String` here, I know. Not my style though :slightly_smiling_face:

Reply:
What's wrong with `show $ typeRep (Proxy @a)` though? 😄

Reply:
This type annotation is redundant here.

Reply:
Nice, I didn't know it would be so easy, changed :+1: